### ISSUE TITLE
There was an addtional div when LogoControl was removed

### ISSUE THREAD

**Issue by jingsam** [mapbox/mapbox-gl-js#4813]
**mapbox-gl-js version**:
0.37.0
### Steps to Trigger Behavior

https://jsfiddle.net/jingsam/ue99mkgy/1/

### Expected Behavior

there should not exsit a `mapbox-ctrl` div

### Actual Behavior

![_20170609182456](https://user-images.githubusercontent.com/1522494/26971693-d91e5db6-4d40-11e7-80ee-81788410d4f9.png)

**Comment by jfirebaugh** [comment]
Is this causing a particular issue? The logo control is always preset by default; whether it shows anything is dependent on what style you load and your Mapbox account.

**Comment by jingsam** [comment]
Noop, but I would expect the logo control to be hidden when there were no mapbox sources.

After I checked the [logo_control.js](https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/control/logo_control.js), I found the implemention was rather buggy. For example:
https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/control/logo_control.js#L26
`this._updateLogo()` would do nothing because the parameter `e` is undefined

https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/control/logo_control.js#L47
I think it should be `this._map.off('sourcedata', this._updateLogo);` Another question is why close the `sourcedata` event? If users remove all mapbox sources and then readd mapbox sources, `this._updateLogo` may not fire again.

The logic of `_updateLogo` is also problematic. When `this._logoRequired()` is false, the logoControl should be hidden but not to be removed.

I think a simpler and cleaner logic to implement the logoControl is: creating all necessary div and a tag in `onAdd`, and then if there exists any mapbox sources, display the control; if not, hide this control with `display: none`
